### PR TITLE
Split out luarocks.vers from luarocks.deps

### PR DESCRIPTION
**Pull request by mpeterv**
luarocks.deps module used to contain version and dependency parsing
functions as well as some high-level dependency handling:
matching and resolving them. A lot of modules only need the parsing
half. Split it into its own module, luarocks.vers. Also remove a workaround for a cyclic dependency
between fetch, deps and manif.

### CODE REVIEW

File: src/luarocks/deps.lua [hunk 1]
```diff
@@ -1,147 +1,13 @@
...
--- comparison criteria is the source code of this module, but the
--- test/test_deps.lua file included with LuaRocks provides some
--- insights on what these criteria are.
+--- High-level dependency related functions.
 local deps = {}
-setmetatable(deps, { __index = require("luarocks.core.deps") })
 
 local cfg = require("luarocks.core.cfg")
-local manif = require("luarocks.core.manif")
+local manif = require("luarocks.manif")
```
Comment (hishamhm) [inline src/luarocks/deps.lua]: are you going to move get_versions from luarocks.core.manif to luarocks.manif?

Comment (mpeterv) [inline src/luarocks/deps.lua]: Oh yes, I forgot about the thing that was the initial reason for all this :smile: Yes, I'll move it. Inside or outside this PR, I don't think it matters.

Comment (hishamhm) [inline src/luarocks/deps.lua]: please do it in this PR so we document in the history why the change was made :)

Comment (hishamhm) [inline src/luarocks/deps.lua]: nevermind, I can move it and just reference this PR in the commit!
